TeachingRecordSystem/src/TeachingRecordSystem.SupportUi/PageModelExtensions.cs

```diff
@@ -1,8 +1,45 @@
+using System.Transactions;
 using Microsoft.AspNetCore.Mvc.RazorPages;
+using TeachingRecordSystem.Core.Services.TrsDataSync;
 
 namespace TeachingRecordSystem.SupportUi;
 
 public static class PageModelExtensions
 {
+    private static readonly TimeSpan _defaultSyncThreshold = TimeSpan.FromSeconds(5);
+
     public static PageResult PageWithErrors(this PageModel pageModel) => new PageResult() { StatusCode = StatusCodes.Status400BadRequest };
+
+    public static async Task SyncPersonAsync(
+        this PageModel pageModel,
+        Guid personId,
+        CancellationToken cancellationToken = default)
+    {
+        var syncHelper = pageModel.HttpContext.RequestServices.GetRequiredService<TrsDataSyncHelper>();
+
+        using var txn = new TransactionScope(TransactionScopeOption.Suppress, TransactionScopeAsyncFlowOption.Enabled);
+        await syncHelper.SyncPersonAsync(personId, syncAudit: false, cancellationToken: cancellationToken);
```

Comment:
Just curious, is the transaction aborted if the token is cancelled?

Reply:
Also `SyncPersonAsyc()` creates its own transaction, assuming that's not an issue?

Reply:
https://github.com/DFE-Digital/teaching-record-system/blob/fe62255374dca7f7e20fb15cb88d8a4be4a313a3/TeachingRecordSystem/src/TeachingRecordSystem.Core/Services/TrsDataSync/TrsDataSyncHelper.cs#L487

Docs for `BeginTransactionAsync()` say nested transactions are not supported

Reply:
The `TransactionScopeOption.Suppress` is there to avoid the nested transactions problem; it's fine that it doesn't share a parent transaction.

> Just curious, is the transaction aborted if the token is cancelled?

The transaction created inside`TrsDataSyncHelper` would be, yes.